src/DTO/Bandit/ActionCoefficients.php

```diff
@@ -0,0 +1,75 @@
+<?php
+
+namespace Eppo\DTO\Bandit;
+
+use Eppo\Exception\InvalidArgumentException;
+
+class ActionCoefficients
+{
+    /**
+     * @param string $actionKey
+     * @param float $intercept
+     * @param NumericAttributeCoefficient[] $subjectNumericCoefficients
+     * @param CategoricalAttributeCoefficient[] $subjectCategoricalCoefficients
+     * @param NumericAttributeCoefficient[] $actionNumericCoefficients
+     * @param CategoricalAttributeCoefficient[] $actionCategoricalCoefficients
+     * @throws InvalidArgumentException
+     */
+    public function __construct(
+        public readonly string $actionKey,
+        public readonly float $intercept,
+        public readonly array $subjectNumericCoefficients = [],
+        public readonly array $subjectCategoricalCoefficients = [],
+        public readonly array $actionNumericCoefficients = [],
+        public readonly array $actionCategoricalCoefficients = []
+    ) {
+        // Since PHP doesn't enforce array typing and this dev spent an unfortunate amount of time debugging, we ensure
```

Comment:
higher levels of DTOs properly enforce the typings

Reply:
I love how the comment alludes to this decision being made from blood sweat and tears 😅